### DISCUSSION TITLE
PDF viewer not working when uploading PDF files.

### DISCUSSION THREAD

**Discussion by Tehnari** [exoplatform/exo-community#20] [Q&A]
Hello!
I have a problem with viewer script for PDF files. With office files all is working as usual. Tested with docker-compose and manual installation on version 6.4.0 and 6.4.1. Any ideas how to fix?

Error is like 

`PreviewNotAvailable

CannotLoadViewerScript`

![image](https://github.com/exoplatform/exo-community/assets/8842002/d20d3ecb-bc2e-462a-b7ef-0bc633ffb5fc)

**Comment by rdenarie** [comment]
Hello, 
Can you confirm that you use eXo Community version ?
In fact, I'm able to reproduce the problem with PDF and with office files.

When opening a pdf file, can you open the js console, and provide logs ? 

Thanks

**Reply by Tehnari** [reply]
Yes, I use community version. Below is browser console log, Exoplatform was started from docker-compose, in this case version 6.4.0. Changed port for nginx from 80 to 9980. Configuring OnlyOffice Document server and exo-onlyoffice addon doesn't help. 

`
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. [rc2Contentscript.js:247:10](moz-extension://a40a9d90-ea7a-4425-9976-14a73731a885/contentscripts/rc2Contentscript.js)
TypeError: document.body is null
[rc2Contentscript.js:249:29](moz-extension://a40a9d90-ea7a-4425-9976-14a73731a885/contentscripts/rc2Contentscript.js)
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. [documents](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents)
GET
http://localhost:9980/eXoSkin/skin/img/glyphicons-halflings.png
[HTTP/1.1 404  1ms]

| TRACE | [webconferencing_966128] < Loaded at http://localhost:9980/portal/g/:spaces:test_space/test_space/documents -- 2023-06-13T12:27:47.078Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [webrtc*_966128] < Loaded at http://localhost:9980/portal/g/:spaces:test_space/test_space/documents -- 2023-06-13T12:27:47.079Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [jitsi*_966128] < Loaded at http://localhost:9980/portal/g/:spaces:test_space/test_space/documents -- 2023-06-13T12:27:47.079Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
GET
http://localhost:9980/digital-workplace/skin/images/eXo.png
[HTTP/1.1 404  1ms]

| TRACE | [webconferencing_966128] Added call provider: jitsi (Jitsi) -- 2023-06-13T12:27:47.314Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [webconferencing_966128] Current user not set, later will try initialized provider: jitsi -- 2023-06-13T12:27:47.314Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [webconferencing_966128] Added call provider: webrtc (WebRTC) -- 2023-06-13T12:27:47.315Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [webconferencing_966128] Current user not set, later will try initialized provider: webrtc -- 2023-06-13T12:27:47.315Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user’s experience. For more help http://xhr.spec.whatwg.org/ [jquery--min.js:182:386](http://localhost:9980/portal/scripts/1.3.4-20230124170014/SHARED/jquery--min.js)
| DEBUG | [webconferencing_966128] User initialized in Web Conferencing: timti. Lang: en-US. Local date: 6/13/2023, 3:27:47 PM. Browser: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0 -- 2023-06-13T12:27:47.479Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| DEBUG | [webconferencing_966128] Initialized call provider: jitsi -- 2023-06-13T12:27:47.480Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| DEBUG | [webconferencing_966128] Initialized call provider: webrtc -- 2023-06-13T12:27:47.480Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
| TRACE | [webconferencing_966128] User updates subscribed successfully: {"channel":"/meta/subscribe","id":"4","subscription":"/eXo/Application/WebConferencing/user/timti","successful":true} -- 2023-06-13T12:27:47.577Z [webConferencingGRP--min.js:27:430](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/webConferencingGRP--min.js)
Loading deffered tasks [cloudDriveGRP--min.js:5:243](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/cloudDriveGRP--min.js)
XML Parsing Error: not well-formed
Location: http://localhost:9980/portal/rest/contentviewer/repository/collaboration/b4b4452dac19000529a64724b0d33cdd
Line Number 180, Column 61: [b4b4452dac19000529a64724b0d33cdd:180:61](http://localhost:9980/portal/rest/contentviewer/repository/collaboration/b4b4452dac19000529a64724b0d33cdd)
GET
http://127.0.0.1/OfficeWeb/apps/api/documents/api.js
[HTTP/1.1 503 Service Unavailable 0ms]

Loading failed for the <script> with source “http://127.0.0.1/OfficeWeb/apps/api/documents/api.js”. [documents:1:1](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents?documentPreviewId=b4b4452dac19000529a64724b0d33cdd)
[onlyoffice] Initialize viewer for document: b4b4452dac19000529a64724b0d33cdd -- 2023-06-13T12:27:53.123Z [onlyofficeGRP--min.js:27:9](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/onlyofficeGRP--min.js)
[onlyoffice] i18n: {} -- 2023-06-13T12:27:53.123Z [onlyofficeGRP--min.js:27:9](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/onlyofficeGRP--min.js)
The resource at “http://localhost:9980/eXoSkin/skin/fonts/vuetify/flUhRq6tzZclQEJ-Vdg-IuiaDsNc.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. [documents](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents?documentPreviewId=b4b4452dac19000529a64724b0d33cdd)
The resource at “http://localhost:9980/eXoSkin/skin/fonts/vuetify/fa-regular-400.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. [documents](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents?documentPreviewId=b4b4452dac19000529a64724b0d33cdd)
The resource at “http://localhost:9980/portal/rest/kudos/api/kudos/3/sent?limit=300&returnSize=true&periodType=&dateInSeconds=0” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. [documents](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents?documentPreviewId=b4b4452dac19000529a64724b0d33cdd)
The resource at “http://localhost:9980/portal/rest/clouddrive/features/status/enabled” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. [documents](http://localhost:9980/portal/g/:spaces:test_space/test_space/documents?documentPreviewId=b4b4452dac19000529a64724b0d33cdd)
[onlyoffice] ERROR: ONLYOFFICE script load timeout: [http://127.0.0.1:80/OfficeWeb/apps/api/documents/api.js](http://127.0.0.1/OfficeWeb/apps/api/documents/api.js) -- 2023-06-13T12:27:56.326Z [onlyofficeGRP--min.js:27:9](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/onlyofficeGRP--min.js)
[onlyoffice] ERROR: editor config creation failed : ONLYOFFICE script load timeout. Ensure Document Server is running and accessible. -- 2023-06-13T12:27:56.326Z [onlyofficeGRP--min.js:27:9](http://localhost:9980/portal/scripts/1.3.4-20230124170014/GROUP/onlyofficeGRP--min.js)
It seems that SpaceTopBannerLogo didn't stopped displaying top bar loading
`

**Reply by rdenarie** [reply]
I'm afraid that the port modification from 80 -> 9980 is the root cause of the problem.

In fact, when loading the preview, we need to access to a file named api.js, served by the onlyoffice container.
As I can see in js logs, this file is not loaded : 
GET
http://127.0.0.1/OfficeWeb/apps/api/documents/api.js
[HTTP/1.1 503 Service Unavailable 0ms]
As it is requested on port 80.

But, this problem should be present for other office file, not only pdf.

You can configure the onlyoffice host by adding this property :
onlyoffice.documentserver.host: 127.0.0.1:9980

For that, create a file on your computer named exo.properties, then mount it in exo container, in /opt/exo/gatein/conf/exo.properties


Can you provide docker-compose file if you still have problem ?

Thanks

**Reply by Tehnari** [reply]
Nope, at the server the same error, but port wasn't modified. I'll try to disconnect or uninstall exo-onlyoffice addon and after that will try.
And at main server documents/api.js is available. And tested in connection/integration with other system.

**Comment by Tehnari** [comment]
I uninstalled the add-on exo-onlyoffice in the main (pre-production) server and the PDF files can be viewed as the other files. Adding property exo.onlyoffice.enabled=false in the exo.properties doesn't work.
Result: In my case, uninstalling the add-on was a solution.

**Reply by rdenarie** [reply]
With this solution, you should not be able to edit office documents in eXo.
Do you confirm ?

**Reply by Tehnari** [reply]
Yes. But in case when I need only to show documents this can be enough.
